windows.md

```diff
@@ -254,6 +254,7 @@ With those compatibility things out of the way, you're ready to start the system
     ```bash
     echo "export PATH=\$PATH:\"/c/Program Files/PostgreSQL/16/bin\"" >> "$USERPROFILE/.bash_profile"
     echo "export PGDATA=\"/c/Program Files/PostgreSQL/16/data\"" >> "$USERPROFILE/.bash_profile"
+    echo "export LC_ALL=en_US.UTF-8" >> "$USERPROFILE/.bash_profile"
```

Comment:
1. does this actually set the environment variable in the correct way on Windows? @ProchaLu mentioned `setx` in https://github.com/upleveled/system-setup/issues/76#issuecomment-2213581187, but if that's proprietary / Windows-specific, we should definitely try to do the `.bash_profile` version
2. same question as in https://github.com/upleveled/system-setup/issues/76#issuecomment-2213768403:
   > what exactly is affected here, from the "full system", when setting `LC_ALL`? is it only the command line? if so, that may be acceptable

Reply:
The last time i also used this `setx`, but I don't know if this `setx` is setting it correctly as the echo export... would

https://github.com/upleveled/system-setup/issues/73#:~:text=Victor%20suggested%20I%20run%20setx%20LC_ALL%20en_US.UTF%2D8%20(as%20administrator)%20on%20the%20command%2Dline%2C%20which%20then%20after%20closing%20and%20opening%20the%20terminal%20again%20running%20locale%20the%20result%20changed%20to

Reply:
The `lc_messages` is responsible for displaying the messages in the `psql` console; when it is set to German `de_AT.UTF-8`,  then the messages like error/success output will be in German. E.g `FEHLER:  Relation »nothing« existiert nicht`.

Even though the `LC_MESSAGES` is set to English in this output below, it doesn't affect `lc_messages` value because `LC_MESSAGES` is `Environment Variable`  while `lc_messages` is `PostgreSQL Configuration Parameter` that lives inside the `postgresql.conf` file.

```bash
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=en_US.UTF-8
```

Testing in Linux

`SHOW lc_messages`

Displays output in German
```bash
 lc_messages 
-------------
 de_AT.UTF-8
```

Displays output in English
```bash
lc_messages 
-------------
 en_US.UTF-8
```

Testing on Windows

`SHOW lc_messages`

Two windows tested display different outputs but indicate all is set to the English language, as shown below

```bash
lc_messages
----------------------
 English_Austria.1252
(1 row)
```
```bash
lc_messages
-----------------------------
 English_United Kingdom.1252
```
When `lc_messages` was set back to `de_AT.UTF-8`, it resulted back to German display outputs

This `lc_messages` needs to be set to `en_US.UTF-8` inside the  `postgresql.conf` file in order to make the setting persist across postgres sessions

Reply:
this still needs a full answer

Reply:
this still needs a full answer to all 3 points above

Reply:
After all the investigations, it is determined that we don't need to update the `locale` but rather the setting inside the `postgres.conf` file, which would change stuff for Postgres alone and not the entire system

Reply:
one last thing is still open here, point 3 from [the comment above](https://github.com/upleveled/system-setup/pull/80#discussion_r1668469385):

3) same comment as in  https://github.com/upleveled/system-setup/issues/76#issuecomment-2213768403 :
   > keep in mind that we already set `LC_ALL` on macOS:
   > - https://github.com/upleveled/system-setup/blob/3c6a6084e99d21be9165418f0dbbcf82084dd71e/macos.md?plain=1#L193
   > - https://github.com/upleveled/system-setup/commit/21a1cd8f14358b884e9fa3c31e8526efd4eedf92
   >
   > [if we will apply this for Windows and Linux too,] ... make sure that we do it all the same way in all of Windows, macOS, Linux


some new questions, now that we have a `perl` cross-platform solution:

- what is the point of setting `LC_ALL` if we are already editing the `postgresql.conf` file?
- is it for the case that the `postgresql.conf` file does not contain the `lc_messages` line?
- if it's for that case, do we want to avoid that case? eg. always configure the language via `lc_messages` in `postgresql.conf`? (even if there is not already a configuration value in that file?)

Reply:
> what is the point of setting LC_ALL if we are already editing the postgresql.conf file?

This `LC_ALL` and others like it are the `locale` setting which is for the entire system. 

```bash
~ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```
This ones are the ones that are specific to `Postgres` and setting `LC_ALL` does not affect these ones because your `locale` could be in English but the ones inside the `postgresql.conf` file will still be having the default value and would also affect how postgres work. Anytime you run `SHOW <variable name>` in `psql` the value it shows are the ones inside the `postgresql.conf` file. So I don't think the above directly affects how psql displays things or function.
```bash
lc_messages = 'C'
lc_monetary = 'C'
lc_numeric = 'C'
lc_time = 'C'
```
But the issue then arises when you have this kind of setting which is what cornelia had `system locale: LC_MESSAGES="en_US.UTF-8"` => `Postgres locale: lc_messages='de_AT.UTF-8'`. Mostly after installations the values for the `Postgres locale should be "C"` which simply means use default but for whatever reason hers was set to german Austria.

So, to avoid this kind of conflict in the future, I still believe that we should still be setting the `LC_ALL` to English like we already do for some system and also set this `lc_messages` from the `postgresql.conf` file to the same English as well

> is it for the case that the postgresql.conf file does not contain the lc_messages line?

No it is not, after Postgres installations. These `lc_messages, lc_monetary, lc_numeric, lc_time` variables are found inside the `postgresql.conf` file. I don't think there is a case where they would be missing

> if it's for that case, do we want to avoid that case? eg. always configure the language via lc_messages in postgresql.conf? (even if there is not already a configuration value in that file?)

We can do this, but this variables should exists after successful Postgres installation

Reply:
We decided to remove the `LC_ALL` setting and set the `lc_messages` and `lc_time` instead as these are sufficient and there is no need to set `LC_ALL`

8c90f7b265612699633b3f362230a81f4924a977
https://github.com/upleveled/system-setup/pull/80/commits/e46246077bc7a328a09fd8043c347a2907ae04c5